pkg/controller/jivavolume/jivavolume_controller.go

```diff
@@ -0,0 +1,702 @@
+							fmt.Sprintf(svcNameFormat, cr.Name, cr.Namespace),
+							"--size",
+							fmt.Sprint(capacity),
+							"openebs",
+						}).
+						WithImagePullPolicy(corev1.PullIfNotPresent).
+						WithPrivilegedSecurityContext(&prev).
+						WithResources(&cr.Spec.ReplicaResource).
+						WithVolumeMountsNew([]corev1.VolumeMount{
+							{
+								Name:      "openebs",
+								MountPath: "/openebs",
+							},
+						}),
+				),
+		).
+		WithPVC(
+			pvc.NewBuilder().
+				WithName("openebs").
+				WithNamespace(cr.Namespace).
+				WithOwnerReferenceNew([]metav1.OwnerReference{{
+					APIVersion:         cr.APIVersion,
+					BlockOwnerDeletion: &blockOwnerDeletion,
+					Controller:         &controller,
+					Kind:               cr.Kind,
+					Name:               cr.Name,
+					UID:                cr.UID,
+				},
+				}).
+				WithStorageClass(cr.Spec.ReplicaSC).
+				WithAccessModes([]corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}).
+				WithCapacity(fmt.Sprint(capacity)),
+		).Build()
+
+	if err != nil {
+		return fmt.Errorf("failed to build statefulset object, err: %v", err)
+	}
+
+	instance := &appsv1.StatefulSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: stsObj.Name, Namespace: stsObj.Namespace}, instance)
+	if err != nil && errors.IsNotFound(err) {
+		// Set JivaVolume instance as the owner and controller
+		if err := controllerutil.SetControllerReference(cr, stsObj, r.scheme); err != nil {
+			return err
+		}
+
+		reqLog.V(2).Info("Creating a new Statefulset", "Statefulset.Namespace", stsObj.Namespace, "Sts.Name", stsObj.Name)
+		err = r.client.Create(context.TODO(), stsObj)
+		if err != nil {
+			return err
+		}
+		// Statefulset created successfully - don't requeue
+		return nil
+	} else if err != nil {
+		return operr.Wrapf(err, "failed to get the statefulset details: %v", stsObj.Name)
+	}
+
+	return nil
+}
+
+func updateJivaVolumeWithServiceInfo(r *ReconcileJivaVolume, cr *jv.JivaVolume, reqLog logr.Logger) error {
+	ctrlSVC := &v1.Service{}
+	if err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      cr.Name + "-jiva-ctrl-svc",
+			Namespace: cr.Namespace,
+		}, ctrlSVC); err != nil {
+		return fmt.Errorf("%s, err: %v", updateErrMsg, err)
+	}
+	cr.Spec.ISCSISpec.TargetIP = ctrlSVC.Spec.ClusterIP
+	var found bool
+	for _, port := range ctrlSVC.Spec.Ports {
+		if port.Name == "iscsi" {
+			found = true
+			cr.Spec.ISCSISpec.TargetPort = port.Port
+			cr.Spec.ISCSISpec.Iqn = "iqn.2016-09.com.openebs.jiva" + ":" + cr.Spec.PV
+		}
+	}
+
+	if !found {
+		return fmt.Errorf("%s, err: can't find targetPort in target service spec: {%+v}", updateErrMsg, ctrlSVC)
+	}
+
+	reqLog.V(2).Info("Updating JivaVolume with iscsi spec", "ISCSISpec", cr.Spec.ISCSISpec)
+	cr.Status.Phase = jv.JivaVolumePhasePending
+	if err := r.client.Update(context.TODO(), cr); err != nil {
+		return fmt.Errorf("%s, err: %v, JivaVolume CR: {%+v}", updateErrMsg, err, cr)
+	}
+
+	if err := r.getJivaVolume(cr); err != nil {
```

Comment:
Why we need a get here, is it required ?

Reply:
yes